ecosystem/typescript/sdk/src/plugins/ans_client.ts

```diff
@@ -92,8 +98,10 @@ export class AnsClient {
   async getAddressByName(name: string): Promise<string | null> {
     const { domain, subdomain } = name.match(namePattern)?.groups ?? {};
     if (!domain) return null;
-    if (subdomain) return this.getAddressBySubdomainName(domain, subdomain);
-    return this.getAddressByDomainName(domain);
+    const registration = subdomain
+      ? await this.getRegistrationForSubdomainName(domain, subdomain)
+      : await this.getRegistrationForDomainName(domain);
+    return registration === null ? null : registration.target;
```

Comment:
```suggestion
    return registration?.target;
```
if `registration` is `null`, it would return `null`.

Reply:
I'll try it again, but that was creating an undefined :(

Reply:
ochh